### PR TITLE
PAINTROID-315 Refactor Perspective to Kotlin

### PR DESCRIPTION
**Pull request by CodeChamp-SS**
The file that was refactored: Perspective

https://jira.catrob.at/browse/PAINTROID-315

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Paintroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [x] Post a message in the *#paintroid* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: Paintroid/src/main/java/org/catrobat/paintroid/ui/Perspective.kt [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+import kotlin.jvm.Synchronized
+import kotlin.math.max
+import kotlin.math.min
+
+const val MIN_SCALE = 0.1f
+const val MAX_SCALE = 100f
+private const val SCROLL_BORDER = 50f
+private const val BORDER_ZOOM_FACTOR = 0.95f
+
+open class Perspective(bitmapWidth: Int, bitmapHeight: Int) {
```
Comment (ldirry) [inline Paintroid/src/main/java/org/catrobat/paintroid/ui/Perspective.kt:34]: I think its better if we make these private var and keep it as Int

File: Paintroid/src/main/java/org/catrobat/paintroid/ui/Perspective.kt [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+    var scale: Float
+        get() = surfaceScale
+        set(scale) {
+            surfaceScale = max(MIN_SCALE, min(MAX_SCALE, scale))
+        }
+
+    val scaleForCenterBitmap: Float
+        get() {
+            val screenSizeRatio = surfaceWidth.toFloat() / surfaceHeight
+            val bitmapSizeRatio = bitmapWidth / bitmapHeight
```
Comment (ldirry) [inline Paintroid/src/main/java/org/catrobat/paintroid/ui/Perspective.kt:76]: here we would need to cast to float then

File: Paintroid/src/main/java/org/catrobat/paintroid/ui/Perspective.kt [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+            ratioDependentScale = max(ratioDependentScale, MIN_SCALE)
+            return ratioDependentScale
+        }
+
+    private var bitmapWidth: Float
+    private var bitmapHeight: Float
+    private var initialTranslationX = 0f
+
+    init {
+        this.bitmapWidth = bitmapWidth.toFloat()
```
Comment (ldirry) [inline Paintroid/src/main/java/org/catrobat/paintroid/ui/Perspective.kt:92]: but then we can also remove this init block and other occurences where we cast `bitmapWidth` and `bitmapHeight` to float

**Review by ldirry — APPROVED** [review]
Well done. Just a small remark :)

Edit: Ah oops didnt want to approve yet, but please implement this then I move to PO review ^^

**Comment by CodeChamp-SS** [comment]
> Well done. Just a small remark :)
> 
> Edit: Ah oops didnt want to approve yet, but please implement this then I move to PO review ^^

sure, I'll just implement those changes :)

File: Paintroid/src/main/java/org/catrobat/paintroid/ui/Perspective.kt (hunk 1 above)
Comment (CodeChamp-SS) [reply Paintroid/src/main/java/org/catrobat/paintroid/ui/Perspective.kt]: done :)

**Pull request merged**